crates/ide-assists/src/handlers/bool_to_enum.rs

```diff
@@ -461,7 +461,17 @@ fn add_enum_def(
     usages: &UsageSearchResult,
     target_node: SyntaxNode,
     target_module: &hir::Module,
-) {
+) -> Option<()> {
+    if ctx
```

Comment:
Instead of having this check by syntactic we ought to do it semantically, we can fetch the desf of the relevant module via `semantics.scope(&insert_before)?.module().scope(None)`, and then search through that for the `Bool` name. `insert_before` being the `let insert_before = node_to_insert_before(target_node);` local a few lines below